tables/Tables/ConcatTable.cc

```diff
@@ -319,14 +319,10 @@ namespace casacore { //# NAMESPACE CASACORE - BEGIN
   void ConcatTable::initialize()
   {
     // Check if all tables have the same description.
-    // Note that we size the table instead of reserve, because push_back
-    // gives the following warning for CountedPtr:
-    //  "dereferencing pointer aonymous  does break strict-aliasing rule"
-    vector<CountedPtr<TableDesc> > actualDesc(tables_p.nelements());;
+    std::vector<std::shared_ptr<TableDesc>> actualDesc(tables_p.nelements());;
```

Comment:
Optional: With `std::shared_ptr`, using `push_back()` should be possible again.

Reply:
Knowing the vector size in advance is good; it avoids resizing. vector.reserve could be used with push_back, but sizing the vector with the very cheap shared_ptr constructor seems ok.